investment-app/TRINV.Api/TRINV.Application/Queries/Dashboard/GetUserInvestmentsPerformanceListQuery.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace TRINV.Application.Queries.Dashboard;
+
+using DigitalCurrency;
+using Domain.Entities;
+using Domain.Enums;
+using Interfaces;
+using MediatR;
+using Shared.Business.Utilities;
+
+public record GetInvestmentPerformanceQuery : IRequest<OperationResult<List<InvestmentPerformanceQueryModel>>>;
+
+
+internal class GetInvestmentPerformanceQueryHandler : IRequestHandler<GetInvestmentPerformanceQuery, OperationResult<List<InvestmentPerformanceQueryModel>>>
+{
+    readonly IRepository<Investment> _investmentRepository;
+    readonly IUserContext _userContext;
+    readonly IMediator _mediator;
+
+    public GetInvestmentPerformanceQueryHandler(IRepository<Investment> investmentRepository, IUserContext userContext, IMediator mediator)
+    {
+        _investmentRepository = investmentRepository;
+        _userContext = userContext;
+        _mediator = mediator;
+    }
+
+    public async Task<OperationResult<List<InvestmentPerformanceQueryModel>>> Handle(GetInvestmentPerformanceQuery request, CancellationToken cancellationToken)
+    {
+        var operationResult = new OperationResult<List<InvestmentPerformanceQueryModel>>();
+
+        var currentUserInvestments = await _investmentRepository
```

Comment:
Did you test it is it work correctly ? 
If not Change the _userContext.userId to 1 and add investments with userId 1.

Reply:
Yes, it is tested!